roles/docker/tasks/main.yml

```diff
@@ -22,4 +22,6 @@
 
 - name: restart docker
   shell: systemctl daemon-reload && systemctl stop docker && systemctl start docker-tcp.socket && systemctl start docker
-  when: docker_env_conf_result.changed or docker_tcp_conf_result.changed or docker_svc_conf_result.changed
+  when: docker_install_result.changed or swarm_install_result.changed or docker_env_conf_result.changed or
+        docker_tcp_conf_result.changed or docker_svc_conf_result.changed or force_restart
```

Comment:
Yes, this will be set when the user wants to force a restart regardless of other conditions... e.g. when trying to fix a broken set up.